### ISSUE TITLE
[Feature Request] A Linux version

### ISSUE THREAD

**Issue by mhsabbagh** [ValdikSS/GoodbyeDPI#62]
Linux support by this tool would be great.

**Comment by r4sas** [comment]
Please read [README](README.md) from start to end before requesting such thing.

**Comment by mhsabbagh** [comment]
Good, anything else?

**Comment by sgeto** [comment]
Short version: it's possible, but it would take an entire rewrite.

**Comment by r4sas** [comment]
https://github.com/ValdikSS/GoodbyeDPI/blob/master/README.md#similar-projects
specially for u.

**Comment by mhsabbagh** [comment]
@r4sas Which redirects to another different project, which is completely unrelated to what am asking although I just used 10 words to easily describe my request.

**Comment by ilkkaya** [comment]
linux wiki please

**Comment by rezad1393** [comment]
I cant even read the zagret readme file.
it is all in russian.
can it be translated or at least described by a russian guy, how can I use that application?
it is the same as this one and I just run and also add iptable rules?

**Comment by r4sas** [comment]
@rezad1393 please dont post off topic messages.
For any issues with za*p*ret go to https://github.com/bol-van/zapret/issues

> it is the same as this one and I just run and also add iptable rules?

Yes. But GDPI uses WinDivert to create iptables-like rules on windows.

**Comment by rezad1393** [comment]
sorry.
I got sorta desperate.
did I get that right that you are trying to make this app for linux too? if so, where can I track that?

**Comment by Atavic** [comment]
For Linux there's the other project, close to this one. I also can't read russian, but asking for info in a friendly manner may help.

**Comment by Hi-Angel** [comment]
@rezad1393 the top of the readme you're referring to mentions a file `docs/readme.eng.txt`

**Comment by NotVeryTastyCake** [comment]
> For Linux there's the other project, close to this one. I also can't read russian, but asking for info in a friendly manner may help.

Where is this project?

**Comment by Hi-Angel** [comment]
> > For Linux there's the other project, close to this one. I also can't read russian, but asking for info in a friendly manner may help.
> 
> Where is this project?

[Here](https://github.com/bol-van/zapret)

**Comment by NotVeryTastyCake** [comment]
Thank you!

**Comment by devsecopsfx** [comment]
# Issue #62 — IA / UX comprehension packet

Publication shape: six indexed Issue #62 comments. This is text-only design reasoning; it does not authorize implementation. Sources: Issue #64 handoff comment 4983199295; Issue #62 comments 4982904579 and 4983130058; Draft PR #63 comment 4982907165; Issue #64 body and comment 4983133269. PR #63 remains the Draft candidate at head `723375a13b3ee7c2e523d0de5453826c87ac7281` over base `9902d53d859469974153d02c53f3d6472ecc3bc3`; PR #56 remains Draft at head `9902d53d859469974153d02c53f3d6472ecc3bc3`. These bindings are handoff evidence, not a claim of fresh GitHub readback from this worker.

Heartbeat: completed the source reread and fresh GitHub revalidation pass before publishing. Verified live GitHub state: Issue #62 open, Issue #64 closed, PR #56 404, PR #63 404, so the PR/SHA bindings remain unverified handoff evidence only. Remaining work: publish the packet sections in order and confirm each readback, keeping the final comment tail marker exact.

## [1/6] System interpretation, authority, and relationship model

### Product purpose

ServStack/HFA is an Owner-facing control plane over Hermes. It turns conversation and project context into understandable, explicitly governed work while preserving Hermes as the runtime authority. Its responsibilities are presentation, cross-object navigation, project context, truthful state labels, authentication/redaction, and review gates—not replacement Task, Run, Profile, approval, browser, session, or memory engines.

BrowserAgent contributes a browser-work workspace profile inside that shell: it organizes requirements, live browser inspection, prototypes, video, evidence, and comparison around the current Project/Journey/Topic. It is not a second app shell because a second shell would duplicate global navigation, HFA conversation, state labels, authority controls, and responsive behavior. BrowserAgent changes the capability arrangement and defaults; it does not change who owns operational truth.

### Canonical objects and boundaries

| Object | Job and contents | Canonical source / authority | Lifecycle and relationships | View/adapter rule |
|---|---|---|---|---|
| Project | Defines product outcome, scope, owners, repositories, journeys | ServStack project/domain record; Owner governs scope | active → paused → archived; contains Journeys | Overview is a view, not a second Project store |
| Journey | Explains an outcome through stages/milestones | ServStack project context; Owner governs meaning | planned → active → completed/abandoned; belongs to Project | Timeline/cards adapt one Journey record |
| Topic | Focuses current discussion/work theme | ServStack context linked to Project/Journey | open → resolved/archived; links Wishes, Requirements, Tasks | Filter/chip is not canonical state |
| HFA conversation | Conversational entry, interpretation, clarification, links | Hermes session/conversation is transcript truth; user controls intent | open → continued/compressed/closed; may produce a Wish/Proposal link | Chat is not Task, Approval, or execution authority |
| Wish | Captures desired outcome before commitment | ServStack domain record if/when persisted; Owner owns intent | draft → clarified → withdrawn/promoted | A chat answer may remain only conversational; persistence must be explicit/truthful |
| Proposal | Presents interpretation, options, assumptions, impact | ServStack review object linked to origin conversation/Wish | draft → revised → accepted/rejected | Non-executing; cannot claim a Kanban Task exists |
| Requirement | Defines accepted behavior/constraints/acceptance criteria | Accepted requirement set; Owner controls scope changes | draft → in review → accepted → superseded | Requirement views may annotate, never silently rewrite accepted scope |
| Agent / Profile | Identifies persistent executor capability, tools, routing, restrictions | Hermes Profile owns persistent Agent identity | available/restricted/offline; selected by Task assignment | ServStack reads Profile truth; it must not create shadow Agents |
| Task | Durable unit of governed work | Hermes Kanban owns Task identity/state/dependencies | todo/ready/running/blocked/done/archived | Board/detail are read/write adapters to Hermes, never parallel Tasks |
| Run / Attempt | One execution claim against a Task, with outcome and heartbeat | Hermes Kanban/session runtime owns Attempt and heartbeat truth | claimed/running → completed/blocked/crashed/reclaimed/timed out | ACTIVE requires fresh heartbeat and concrete activity; stale fails closed |
| Approval / Decision | Records whether a named action may proceed | Hermes native runtime approval/policy for tool actions; Owner/GitHub governance for product merge/deploy decisions | pending → approved/rejected/expired/revoked | UI aggregates the correct authority; it does not invent a second approval engine |
| Knowledge / Document / Wiki | Durable domain/process meaning and reusable context | Canonical repo/wiki/document source; mutation by its governed authoring flow | draft → reviewed → canonical → stale/superseded | Search/graph/preview are views; Hermes compact memory is not durable project Knowledge |
| Evidence / Audit | Immutable-enough proof of what happened and independent assessment | Hermes logs + Kanban events for runtime; Git/PR/CI for repo change; captured artifacts for visual claims | collected → verified/disputed → superseded, never overwritten | Evidence is not Knowledge authoring and audit is not implementation authority |
| Browser | Live route/DOM/console/session inspection | Live target and browser session are truth for observed UI; mutation still governed | disconnected/loading/ready/error/permission | Browser surface is an adapter; screenshot alone is not route authority |
| Prototype | Reviewable derivative of Proposal/Requirements | Versioned prototype artifact + exact source/candidate binding | draft → review → revised → accepted/retired | Never replaces accepted Requirements or production truth |
| Video | Time-based preview/evidence linked to context | Versioned media artifact and metadata | absent/loading/ready/error/closed/reopened | Player is a view; close/reopen must preserve source relationship and not imply deletion |
| Workspace | Scope/container choosing context and capability profile | ServStack owns workspace composition/context; Hermes owns underlying runtime objects | selected/unselected/disconnected/permission denied | BrowserAgent is one workspace profile, not another shell |
| Surface | A window/panel onto one or more canonical objects | ServStack owns layout/navigation state only | open/focused/split/maximized/closed/error | Surface state never becomes object authority |

### Refined relationship model

```text
Project ─contains→ Journey ─focuses→ Topic
   │                         ├─links→ Requirement set
   │                         └─scopes→ HFA conversation
   │                                      │
   │                        interpret, never execute silently
   │                                      ↓
   └────────────context────────────→ Wish / Proposal ─derives→ Prototype
                                          │
                                 Owner accepts requirements
                                          ↓
                                  explicit “Create Task”
                                          ↓
Hermes Profile ←assigned/delegated──── Hermes Kanban Task
                                          │
                                     claims/creates
                                          ↓
                               Run / Attempt + heartbeat
                                  │               │
                             activity/logs     candidate/artifacts
                                  └──────┬────────┘
                                         ↓
                           Evidence bundle + independent Audit
                                         ↓
                         Owner Decision (revise / authorize next gate)
                             │                 │
                        merge gate        deployment gate

Knowledge / Manuals / Wiki ─context→ Conversation, Requirements, Profile, Task
Evidence ─proves→ Attempt/candidate/decision; it does not become Knowledge by default
Browser / Prototype / Video ─present or verify→ candidate; none owns Task or approval state
```

Refinements from the suggested model:
1. Requirement belongs to governed scope and may be linked from Topic/Proposal; it is not automatically downstream of every chat message.
2. Approval is attached to a named action and authority, not a generic step between Task and implementation.
3. Prototype can precede Task as design review and can also be an implementation candidate afterward; its version and parent link disambiguate those roles.
4. Evidence is produced throughout an Attempt and review, not only after “implementation.”
5. Merge and deployment are two decisions, not one terminal arrow.

---

**Comment by devsecopsfx** [comment]
## [2/6] Complete workspace/menu/capability architecture

### Placement vocabulary

- Top-level workspace switcher: changes the broad job/context, never creates an object.
- Overview tab: current Project/Journey/Topic and truthful status.
- Capability launcher: grouped user-facing destinations; replaces technical `New surface`.
- Primary tab: common Owner workflow that must stay one action away.
- Drawer/inspector: supporting context/evidence without losing the current object.
- Advanced group: developer/operator tools, discoverable but not dominant.

Status terms below are deliberately provisional: `existing visible`, `existing generic/hidden`, `fixture/candidate-only`, `missing owner flow`, or `unknown—verify`. They describe current review evidence, not backend capability claims.

Responsive key: D = desktop, T = tablet, M = mobile.

| Entry / status | User job; contents | Source of truth; read/write authority | Start → next | States | D/T/M behavior | Placement rationale |
|---|---|---|---|---|---|---|
| Historical PRs / existing workspace | Review precedent, diffs, outcomes | GitHub PR/Git/CI; read-only here, GitHub governs writes | switcher → PR detail/Compare/Evidence | empty/loading/error/permission/stale | D list+detail; T list→detail; M cards→detail | Global workspace because precedent spans projects; not live runtime authority |
| Hermes Compatibility / existing workspace | See version/capability compatibility and degraded truth | Installed Hermes/runtime evidence + compatibility records; read-only Owner view | switcher → capability evidence/manual | checking/compatible/degraded/unknown/error | D matrix; T grouped rows; M summary+drilldown | Global diagnostic workspace, not BrowserAgent content |
| Manuals / existing workspace | Find operating guidance | Canonical manuals/wiki; governed document authoring elsewhere | switcher/search → document/related capability | empty/loading/stale/error/permission | D tree+doc; T drawer+doc; M search→doc | Global Knowledge route because instructions cross workspaces |
| Design System / existing workspace | Review shared components/tokens/states | Canonical design-system source; code review governs mutation | switcher → component specimen/Compare | loading/version mismatch/error | D catalog; T grid; M single specimen | Global because it governs all shells, not a BrowserAgent tool |
| BrowserAgent / existing candidate workspace profile | Perform browser-centered project work | ServStack workspace composition; underlying objects retain their owners | switcher → Overview/launcher | disconnected/loading/ready/error/permission | D shell; T compact shell; M one surface | Top-level profile, not second shell |
| Overview / Project & Journey / missing clarity in candidate | Understand current Project/Journey/Topic and truth labels | ServStack project context; Owner governs domain links | BrowserAgent default → Requirements/Tasks/Knowledge | no context/loading/ready/stale/disconnected/permission | D context rail+summary; T summary+drawer; M context header+sheet | Default primary tab: orientation precedes tools |
| Requirements / existing generic or fixture; ownership to verify | Review accepted scope, revisions, acceptance criteria | Accepted requirement source; Owner governs acceptance | Overview/chat/proposal → Prototype/Create Task/Compare | empty/draft/review/accepted/superseded/error | D list+detail/split; T tabs; M list→detail | Primary Owner workflow; not hidden under Code |
| Agents / missing discoverable Owner flow | Find Profile capability/status and related Attempts | Hermes Profiles + live routing/status; Profile config is Hermes-governed | launcher/Task assignment → Profile detail/active Runs | none/loading/available/restricted/offline/unknown | D list+inspector; T list→sheet; M cards→detail | Launcher primary group; important but not permanent column |
| Tasks & Runs / missing discoverability though Hermes primitive exists | Track durable work and Attempts/heartbeats | Hermes Kanban and session/run truth; mutations via governed Kanban controls | launcher/context links → Task detail/Attempt/Evidence | empty/ready/running/blocked/done/stale/error | D queue+detail; T queue→detail; M queue→timeline | Primary operations destination; combines navigation, not object identity |
| Approvals / Decisions / missing discoverable Owner flow | Review named action, target, environment, evidence, rollback | Correct native/policy/GitHub/Owner decision authority; adapter only | launcher/task banner → approval detail/Evidence | none/pending/approved/rejected/expired/permission | D inbox+detail; T inbox→detail; M decision cards | Primary review destination; must not create shadow authority |
| Browser / existing candidate surface | Inspect live route, DOM, console and visible state | Live browser/target; changes still gated | launcher/object action → split with Prototype/Evidence | disconnected/loading/ready/error/permission | D full/split; T full then switch; M full-screen with context sheet | Primary build/review tool |
| Prototype / existing candidate surface | Review design/candidate derivative and revisions | Versioned artifact linked to Proposal/Requirement/commit | launcher/conversation/requirement → Compare/Evidence/revision | absent/loading/draft/ready/error/retired | D preview/split; T preview+tabs; M preview+metadata sheet | Primary review tool, always shows parent/version |
| Video / existing candidate surface, public nested behavior unverified | Review motion/flow evidence | Media artifact + exact metadata/binding | launcher/Requirement/Evidence → playback/Compare | absent/loading/ready/error/closed | D player/split; T player; M full-screen; reopen restores link | Primary media capability, closable without deleting artifact |
| Knowledge / Documents / missing discoverable Owner flow | Find durable context relevant to current object | Canonical wiki/docs; governed authoring elsewhere | launcher/contextual “Related” → doc/Graph/Task | empty/loading/stale/error/permission | D results+doc; T list→doc; M search→doc | Primary context destination; remains distinct from Evidence |
| Knowledge Graph / existing generic/hidden or derived; verify | Discover relationships/provenance | Derived graph manifest pointing to canonical docs | Knowledge/launcher → node inspector/document | empty/building/stale/error/permission | D canvas+inspector; T canvas→sheet; M list-first with optional graph | Secondary Knowledge tab because graph is discovery, not canonical reading |
| Evidence / Audit trail / missing discoverable Owner flow | Verify exact Task/Attempt/commit/artifacts and audit findings | Hermes logs/Kanban + Git/PR/CI + captured artifacts; governed append/review | launcher/Task/Approval → evidence detail/Compare/Decision | empty/collecting/verified/disputed/stale/error | D timeline+artifact; T timeline→artifact; M evidence cards | Primary review destination; audit separated from authoring |
| Terminal / existing generic surface | Run or inspect commands within governed scope | Selected Hermes terminal backend/session; command policy governs writes | launcher Advanced → terminal/Evidence | disconnected/connecting/ready/running/error/approval required | D panel; T full surface; M read-mostly/full-screen warning | Advanced: high power, not an Owner default |
| Code / existing generic surface | Inspect repository files/diffs | Git worktree/repo; Git workflow governs mutation | launcher Advanced/Task → file/diff/Compare | no repo/loading/clean/dirty/error/permission | D tree+editor; T tree drawer+viewer; M read-only diff first | Advanced: supports work, does not define product IA |
| Data table / existing generic surface | Inspect structured records/results | Named dataset/read model; owner depends on dataset | launcher Advanced/context action → row detail/Compare | empty/loading/ready/error/truncated/permission | D dense grid; T horizontal/column chooser; M cards | Advanced generic viewer; label must include data identity |
| Compare / existing candidate/generic surface | Explain differences between versions, objects, or evidence | Compared canonical sources/artifacts; read-only composition | any versioned object → side-by-side/diff/Decision | missing side/loading/ready/error/incompatible | D side-by-side; T toggle+diff; M sequential A/B+changes | Context action and launcher utility, not permanent column |

### Capability launcher groups

```text
Open capability
  Understand: Overview, Requirements, Knowledge, Knowledge Graph
  Coordinate: Agents, Tasks & Runs, Approvals / Decisions
  Build & review: Browser, Prototype, Video, Compare
  Verify: Evidence / Audit trail
  Advanced: Terminal, Code, Data table
```

Every important capability remains reachable from: (1) launcher, (2) object-specific related links, and (3) command/search. No capability depends on a permanent column.

### Progressive disclosure and layout reasoning

- Agent, Knowledge, Approval, Tasks, and Evidence cannot disappear because they answer different high-risk questions: who can act, what is known, who authorized what, what work exists, and what proves it.
- They should not all become columns. Columns consume width even when irrelevant, turn objects into “modes,” and make splits unusable. A launcher plus contextual links keeps them one action away.
- `Open capability` is preferable to `New surface`: the user chooses a job/destination; the shell decides whether to focus an existing surface, open a tab, or create a split. “New surface” exposes implementation mechanics and encourages duplicates.
- Owner workflows appear first; developer tools sit in Advanced. Permission and state may hide an action, but the destination remains named with an explanation rather than silently vanishing.
- One shell header carries workspace, context, truth state, launcher, and HFA control. A surface adds only its object title and local actions—no duplicate page/tab/surface chrome.
- Collapsing HFA removes its grid track entirely and returns width to content; it must not leave a blank reserved column.
- Splits require sufficient width. Desktop supports adjustable two-pane layout. Tablet switches between linked tabs or a primary pane plus sheet. Mobile uses one surface at a time with a persistent “paired with…” control; it never squeezes two unreadable panes.
- User-facing labels are Project, Requirements, Agents, Tasks & Runs, Approvals, Knowledge, Evidence, Browser, Prototype, Video, Compare. Technical IDs, SHAs, adapter names, event names, and backend diagnostics stay in Evidence/diagnostic inspectors.

---

**Comment by devsecopsfx** [comment]
## [3/6] Twelve screen-by-screen stories

Each story states start, control, destination, visible data, state/authority, progress/evidence, continuation, and responsiveness.

### S1 — Understand Project / Journey / Topic
Start: BrowserAgent opens on Overview. Control: workspace switcher → BrowserAgent. Opens: Overview. Visible: Project name/scope, active Journey stage, Topic, current requirement/candidate bindings, truthful connected/degraded label. State: selection only; no runtime mutation. Authority: project context comes from ServStack domain records. Progress/evidence: linked Tasks and latest verified evidence summaries. Continue: Requirements, Tasks & Runs, Knowledge, or launcher. D shows context rail + overview; T collapses rail into sheet; M shows compact context header and drilldown.

### S2 — Find an Agent and related active Runs
Start: Overview or launcher. Control: Open capability → Coordinate → Agents. Opens: Agent list; selecting one opens Profile detail. Visible: capability description, status/freshness, restrictions, routing, active/recent Attempts. State: read-only until an explicit governed Profile/config action outside this packet. Authority: Hermes Profile and runtime status. Progress: active Attempts only when heartbeat is fresh; stale shows STALE/UNKNOWN. Evidence: links to run logs/events. Continue: Task/Attempt detail or assignment flow. D list+inspector; T list→sheet; M cards→detail.

### S3 — Follow Task / Run / Attempt progress
Start: Overview, Agent, or Tasks & Runs. Control: linked Task or launcher → Tasks & Runs. Opens: Task detail with Attempt timeline. Visible: scope, origin, assignee, dependencies, state, current Attempt, last heartbeat/activity, prior outcomes. State: Hermes Kanban truth; no inferred ACTIVE. Authority: Kanban mutations only through explicit controls. Progress: Task header and Attempt timeline. Evidence: per-Attempt artifacts/logs, never flattened. Continue: Evidence, Approval, Agent, revision Task. D queue+detail; T queue→detail; M Task summary→timeline.

### S4 — Review an Approval without a second authority
Start: Approval badge on Task or launcher → Approvals. Control: select pending request. Opens: Approval detail. Visible: exact action, target, environment, requester, authority type, evidence, risks, rollback, expiry. State: pending/approved/rejected/expired from the canonical authority. ServStack submits only through the owning Hermes policy or records an Owner/GitHub decision in its proper system; it does not mirror a separate executable approval. Progress: decision history. Evidence: immutable links. Continue: return to Task or next separate gate. D inbox+detail; T list→detail; M review card with evidence before decision controls.

### S5 — Find relevant Knowledge / Wiki
Start: Task/Requirement or launcher. Control: Related Knowledge or Open capability → Understand → Knowledge. Opens: filtered results then canonical document view. Visible: title, authority/status, source path/version, relevance rationale, links. State: navigation/read only; authoring uses governed doc flow. Authority: canonical wiki/docs, not graph or compact memory. Evidence: provenance and freshness, not execution proof. Continue: linked Requirement/Task/Graph. D results+doc; T list→doc; M search/filter→doc.

### S6 — Inspect Evidence / Audit
Start: Task, Attempt, candidate, Approval, or launcher. Control: View evidence. Opens: Evidence/Audit timeline scoped to exact object/version. Visible: Task/Attempt IDs, heartbeat/events, commit/PR/CI, artifact hashes/URLs, browser observations, independent audit verdict and disputes. State: collecting/verified/disputed/superseded; append revisions, never overwrite. Authority: each evidence type names its owner. Continue: Compare, Approval/Decision, prior Attempt. D timeline+artifact inspector; T timeline→artifact; M evidence cards with provenance.

### S7 — Browser and Prototype side by side
Start: Browser or Prototype. Control: Pair/Compare with → Prototype or Browser. Opens: linked split if width permits. Visible: live route truth and prototype version/parent side by side, shared context, independent state labels. State: opening a view only. Authority: live Browser observation and artifact binding remain separate. Evidence: capture control writes a new evidence item linked to both versions. Continue: Compare, report defect, close one pane. D adjustable split; T tab pair or sheet; M one pane with persistent switcher.

### S8 — Requirements + Video; safe close/reopen
Start: accepted/draft Requirement detail. Control: Open related → Video. Opens: Requirement + Video split/tab. Visible: requirement version, media binding, lifecycle state, evidence links. Closing Video removes only the surface; it does not delete media or Requirement relationship. Reopen from Related or launcher restores the same artifact and playback metadata. Authority: requirement and media owners remain separate. D split; T linked tabs; M Video full-screen with return to Requirement.

### S9 — HFA chat asks to improve HFA
Start: any page with HFA conversation open. Control: send “Improve this HFA menu and layout.” Opens: conversational interpretation and a proposed non-executing Wish/Proposal only after explicit save/promotion. Visible: attached context and assumptions. No Task is silently created. Owner reviews Proposal → Requirements → Prototype, then explicitly creates Hermes Task. Progress appears in Tasks & Runs; evidence in Evidence/Audit. Continue through revision, audit, merge gate, deployment gate. Detailed in [4/6]. D side HFA; T overlay/sheet; M full-screen overlay that returns to exact surface.

### S10 — HFA chat asks to improve BrowserAgent
Start: BrowserAgent with HFA open. Control: send request. Opens: response referencing current Project/Journey/Topic and BrowserAgent surface/route. First durable object is Wish/Proposal, not Task. Owner reviews BrowserAgent-specific Requirements and Prototype. Explicit Create Hermes Task selects Profile/delegation. Progress/evidence follows canonical records. D chat beside workspace; T sheet; M overlay with context chip and exact return target.

### S11 — Report a visual defect and request correction
Start: Prototype or Browser/Prototype pair. Control: Report defect / Ask HFA about this view. Opens: HFA composer with route, surface IDs, prototype version, viewport, visible-state capture references; user describes defect. Result: revision request linked to existing Proposal/Requirement/Task/Attempt, not overwrite. Authority: accepted scope changes require Owner review; a correction Attempt requires explicit Task continuation/new Task according to scope. Progress: new Attempt or revision record. Evidence: before/after artifacts remain. D selection + chat; T capture sheet; M annotate/report then return.

### S12 — Review corrected candidate and decide next gates
Start: Task/Attempt completion notification or Approval inbox. Control: Review candidate. Opens: Compare + Evidence/Audit with prior/current prototype/browser/video. Visible: exact Task/Attempt/commit, requirements coverage, test/CI, independent audit, unresolved issues. Owner may request revision, authorize implementation continuation, approve merge, or reject. Merge decision does not deploy; after merge evidence, a separate deployment request names environment and rollback. D compare+evidence; T toggled diff+inspector; M sequential prior/current and separate decision cards.

---

**Comment by devsecopsfx** [comment]
## [4/6] Mandatory trace — “Improve this HFA menu and layout.”

### 1. Capture with automatic context

The message envelope attaches only observable identifiers/state, not guessed meaning:

```text
conversation/session link
workspace = current BrowserAgent or other workspace
route + active surface type/id + open paired surfaces
Project / Journey / Topic identifiers and displayed labels
candidate repository/ref/head if the current surface has a verified binding
viewport/device class
visible truth state (ready/disconnected/loading/error) and capture/evidence reference
```

Sensitive content is redacted and unknown fields say UNKNOWN. A screenshot/DOM reference may be attached as evidence; the UI must not claim that every visual fact was captured if only route metadata exists.

### 2. Non-executing first result

The immediate result is a conversational interpretation. If the Owner chooses `Save as proposal`, ServStack creates a non-executing Wish/Proposal linked to the conversation and context envelope. It is not a Requirement change or Task because wording is broad, acceptance criteria are absent, and execution authority has not been granted. The UI states `No Task created`.

### 3. Proposal outside chat

A Proposal appears in the current Project/Journey/Topic review queue and under the conversation’s linked objects. It contains problem framing, affected surfaces, assumptions, options, out-of-scope items, risks, and requested next review. Chat links to it; the Proposal is independently navigable.

### 4. Requirements and Prototype review

Owner selects `Review requirements`. The Requirement set shows proposed additions/changes against the accepted version. `Open prototype` displays a derivative linked to the Proposal and requirement version. Prototype status never says “implemented” without candidate evidence. Owner can accept, revise, or reject requirements and can request a prototype revision without changing operational state.

### 5. Explicit conversion to Hermes-native Task

Only the control `Create Hermes Task` on an accepted Proposal/Requirement can initiate Task creation. A confirmation shows title, scope, acceptance criteria, origin links, workspace kind, dependencies, assignee choice, and whether execution starts now or waits. The adapter invokes Hermes Kanban; success is shown only after a real Task ID is read back. Failure leaves the Proposal intact and says no Task exists.

### 6. Approval boundary

Approval is required whenever policy or the named action demands it—especially protected commands, repository mutation, merge, deployment, production impact, or scope changes. Proposal acceptance is not runtime execution approval. Task creation is not merge approval. Attempt completion is not deployment approval.

### 7. Profile versus delegation

Assignment UI offers verified persistent Hermes Profiles with capability/status/restriction evidence. Persistent accountable work selects a Profile. Bounded ephemeral reasoning may use Hermes delegation under the parent Task/Attempt when allowed. ServStack does not manufacture a Profile or treat a subagent as persistent identity.

### 8. Task, Attempt, heartbeat

After readback, Task appears in Tasks & Runs and as a linked object in chat/Proposal/Requirement. Claiming creates an Attempt. The detail shows Attempt ID, Profile, start time, last heartbeat, concrete current activity, prior Attempts, and state. If heartbeat exceeds freshness policy or source disconnects, UI stops saying ACTIVE and shows STALE/DISCONNECTED/UNKNOWN.

### 9. Evidence and visual artifacts

Each Attempt owns links to command/test output, Git diff/commit/PR/CI, browser route/DOM/console observations, screenshots, prototype/video versions, and worker handoff. Independent audit adds a separate verdict referencing the exact candidate. Evidence/Audit never rewrites Knowledge and never collapses multiple Attempts into one “latest proof.”

### 10. Open resulting Prototype

From chat: linked object chip `Prototype vN`. From navigation: BrowserAgent → Build & review → Prototype, filtered to current Project/Proposal/Task. Both resolve the same versioned artifact and show parent Proposal/Requirement plus candidate binding.

### 11. Revision without evidence loss

Owner selects `Request revision`, comments against exact artifact/evidence, and chooses whether this is within accepted scope. The system appends a revision request. The next prototype version and Attempt link back to the prior one; old evidence remains read-only and Compare can show vN versus vN+1. If scope changes, Requirements return to review before execution.

### 12. Independent audit, merge, and deployment

A reviewer separate from implementation checks requirements, diff, tests, browser/visual behavior, and evidence provenance. Owner sees worker result and audit result with disagreements explicit. `Authorize merge` is a named GitHub/repository gate tied to exact head and evidence. Deployment is a later, separate request tied to merged artifact, target environment, health checks, rollback, and production authority. Neither button is implied by Task `done`.

### End-to-end state chain

```text
message
  → conversational answer
  → [explicit Save] Wish/Proposal (non-executing)
  → Requirements review
  → linked Prototype review/revisions
  → [explicit Create Hermes Task]
  → real Task readback
  → Profile assignment or bounded delegation
  → Attempt + fresh heartbeat
  → candidate + per-Attempt Evidence
  → independent Audit
  → Owner revision or decision
  → separate merge gate
  → separate deployment gate
```

Failure behavior is fail-closed at every adapter boundary: no readback means no claimed object/state transition.

---

**Comment by devsecopsfx** [comment]
## [5/6] Navigation matrix and low-fidelity wireframes

### Navigation matrix

| # | User intent | Starting location | Control | Destination surface | Canonical object | Next actions |
|---:|---|---|---|---|---|---|
| 1 | Understand current work | workspace switcher | BrowserAgent | Overview | Project/Journey/Topic | Requirements, Tasks, Knowledge |
| 2 | Change Journey/Topic | Overview | context selector | Overview filtered | Project/Journey/Topic | inspect linked work |
| 3 | Review scope | Overview/chat | Requirements | Requirement detail | Requirement set | compare, prototype, accept/revise |
| 4 | Find executor | launcher | Agents | Profile detail | Hermes Profile | active Runs, assign Task |
| 5 | See active work | Overview | Tasks & Runs | queue | Hermes Tasks | Task detail |
| 6 | Follow execution | Task queue | Task row | Task/Attempt detail | Task + Attempt | Evidence, Approval, Agent |
| 7 | Check heartbeat | Attempt detail | activity timeline | Attempt inspector | Hermes Attempt/events | logs, mark stale truthfully |
| 8 | Review pending action | Task/banner | View approval | Approval detail | owning approval/decision record | approve/reject/return |
| 9 | Find relevant guidance | Task/Requirement | Related Knowledge | document results | canonical Knowledge | open doc/Graph |
| 10 | Explore relationships | Knowledge | Graph | Knowledge Graph | derived graph → docs | inspect node/open doc |
| 11 | Verify exact proof | Task/Attempt | View evidence | Evidence/Audit | evidence records | Compare, Decision |
| 12 | Inspect live UI | launcher | Browser | Browser surface | live route/session | pair, capture evidence |
| 13 | Review design | Proposal/Requirement | Open prototype | Prototype | versioned artifact | compare, revision, evidence |
| 14 | Review motion | Requirement/Evidence | Open video | Video | media artifact | close/reopen, annotate |
| 15 | Pair live and design | Browser | Pair with Prototype | split/pair | Browser + Prototype | capture, compare, report defect |
| 16 | Compare revisions | Prototype/Evidence | Compare | Compare | two named versions | explain diff, decide |
| 17 | Ask HFA about view | any surface | open HFA | conversation overlay | Hermes conversation + context refs | save Proposal, continue chat |
| 18 | Save intent | HFA response | Save as proposal | Proposal detail | Wish/Proposal | requirements review |
| 19 | Create durable work | accepted Proposal | Create Hermes Task | Task confirmation/readback | Hermes Task | assign/start/wait |
| 20 | Select persistent executor | Task confirmation | Assign Profile | Agent picker | Hermes Profile | create/start Task |
| 21 | Use bounded subagent | Task detail | Delegate bounded subtask | Attempt activity | Hermes delegation under Task | inspect result/evidence |
| 22 | Report visual defect | Prototype/Browser | Report defect | HFA revision composer | revision request linked to origin | revise requirement/prototype/task |
| 23 | Inspect repository | launcher Advanced | Code | Code surface | Git repo/worktree | diff, Compare, evidence |
| 24 | Inspect structured output | object action | Data table | table | named dataset/read model | row detail/export if allowed |
| 25 | Run governed command | launcher Advanced | Terminal | terminal surface | Hermes terminal session | approval, logs, evidence |
| 26 | Review prior PR | global switcher | Historical PRs | PR detail | GitHub PR | Compare, Evidence |
| 27 | Diagnose compatibility | global switcher | Hermes Compatibility | compatibility matrix | runtime/version evidence | manual/evidence |
| 28 | Review corrected candidate | notification/Task | Review candidate | Compare + Evidence | Attempt/candidate | revise, merge decision |
| 29 | Authorize merge | candidate review | Request/authorize merge | Decision detail | GitHub/Owner decision | merge readback, then deployment request |
| 30 | Authorize deployment | merged candidate | Request deployment | deployment decision | deployment authority | execute/verify/rollback |

### Wireframe A — BrowserAgent default desktop

```text
┌ HFA shell: [Workspace: BrowserAgent] [Project › Journey › Topic] [STATE] [Open capability] [HFA ▸] ┐
├ Context ───────┬ Overview ───────────────────────────────────────────────────────────────────────────┤
│ Project        │ Journey stage / Topic / accepted Requirements / linked Tasks / latest Evidence     │
│ Journey        │ [Review requirements] [Open active task] [Open related knowledge]                  │
│ Topic          │ Truth: source + freshness + candidate binding                                      │
└────────────────┴─────────────────────────────────────────────────────────────────────────────────────┘
```

### Wireframe B — capability launcher open

```text
┌ Open capability ─────────────────────────────┐
│ Understand   Overview  Requirements Knowledge│
│              Knowledge Graph                 │
│ Coordinate   Agents  Tasks & Runs  Approvals │
│ Build/review Browser Prototype Video Compare │
│ Verify       Evidence / Audit trail          │
│ Advanced ▸   Terminal Code Data table        │
│ Search capabilities…                         │
└──────────────────────────────────────────────┘
```

### Wireframe C — Agents

```text
┌ Agents ───────────────┬ Profile detail ─────────────────────────────┐
│ ● factory  fresh      │ Capability / routing / restrictions         │
│ ○ reviewer offline   │ Status source + last observed               │
│ ! worker restricted  │ Active Attempts (fresh only) / recent Runs  │
└───────────────────────┴──────────────────────────────────────────────┘
```

### Wireframe D — Task & Run detail

```text
┌ Task t_xxx [RUNNING] origin: Proposal P-7 ───────────────────────────┐
│ Scope | acceptance | Profile | dependencies | last source refresh   │
├ Attempts ────────────────────────────────────────────────────────────┤
│ A3 RUNNING  heartbeat 17:42  activity: validating menu packet       │
│ A2 BLOCKED  publication auth unavailable  [evidence]                │
│ A1 CRASHED  process lost                    [events]                 │
└ [Evidence] [Approval] [Agent] [Request revision] ────────────────────┘
```

### Wireframe E — Approval detail with evidence

```text
┌ Approval: MERGE exact head abc… [PENDING] ──────────────────────────┐
│ Authority: GitHub/Owner  Target: PR #63  Environment: repository    │
│ Requester / expiry / risks / rollback                               │
├ Evidence ─ tests ─ audit verdict ─ unresolved disagreements ────────┤
│ [Reject] [Request revision] [Approve exact action]                  │
└ Note: approval does not deploy ─────────────────────────────────────┘
```

### Wireframe F — Knowledge entry and document

```text
┌ Related Knowledge ────┬ Canonical document ─────────────────────────┐
│ search/filter          │ title | authority | version | freshness     │
│ result + relevance     │ body / links                                │
│ stale warning          │ [Open source] [Graph] [Linked requirements] │
└────────────────────────┴──────────────────────────────────────────────┘
```

### Wireframe G — Evidence / Audit

```text
┌ Evidence: Task t_x / Attempt A3 / candidate abc… ───────────────────┐
│ 17:41 heartbeat ─ 17:43 tests ─ 17:45 browser capture ─ 17:48 audit │
│ artifact hash / route / DOM-console / PR-CI / reviewer identity      │
│ Audit: FAIL — toolbar overflow      Worker: READY                    │
│ [Compare prior attempt] [Open artifact] [Request revision]           │
└───────────────────────────────────────────────────────────────────────┘
```

### Wireframe H — Requirements + Prototype split

```text
┌ Requirements v4 [IN REVIEW] ───────┬ Prototype v7 [DERIVED] ────────┐
│ changed criteria / acceptance       │ parent Proposal P-7 / Req v4    │
│ assumptions / out of scope          │ candidate binding / truth state │
│ [Accept] [Revise]                   │ [Compare v6] [Report defect]     │
└─────────────────────────────────────┴──────────────────────────────────┘
```

### Wireframe I — mobile BrowserAgent

```text
┌ BrowserAgent [STATE] [⋯] ┐
│ Project › Journey › Topic │
│ Current: Overview         │
│ summary / primary action  │
│ [Open capability]         │
│ paired with Prototype ▸   │
└───────────────────────────┘
```

### Wireframe J — mobile HFA overlay

```text
┌ ← return to Prototype v7 ┐
│ HFA conversation         │
│ Context: BrowserAgent /  │
│ route / Topic / v7       │
│ [inspect attached refs]  │
│ message thread           │
│ composer                 │
│ [Send] [Save proposal]   │
│ “No Task created”        │
└──────────────────────────┘
```

---

**Comment by devsecopsfx** [comment]
## [6/6] Ambiguities, self-check, rejected choices, prototype mistakes, and traceability

### Ambiguity register

| Ambiguity | Current evidence | Interpretations | Recommendation | Consequence if wrong |
|---|---|---|---|---|
| Canonical persistence for Wish/Proposal/Requirement | Issue instructions require objects; Issue #64 forbids duplicate Hermes engines | ServStack domain store, GitHub objects, or future adapter | Treat as ServStack domain concepts with persistence seam explicitly UNKNOWN until Track B/Owner accepts it | UI could claim durable objects that do not exist |
| Product merge/deploy decision record | Hermes owns runtime approval; GitHub owns repo evidence; Owner governs promotion | one generic approval vs authority-specific decisions | One Approvals UI that routes to and labels the owning authority; no generic executable shadow record | duplicate or bypassed authority |
| Agent “live” status | Hermes Profile is persistent identity; runtime liveness may be separate | configured = available, or recent runtime evidence = live | Separate configured capability from observed availability/freshness | false ACTIVE claims |
| Project/Journey/Topic storage | ServStack owns project context, exact schema not locked | repo docs, DB, Issue metadata | Show source/version and keep adapter replaceable; no guessed write behavior | cross-links drift or lose auditability |
| Prototype before versus after Task | Both design review and implementation candidate are needed | one mutable prototype, or versioned roles | Version prototypes and label `design derivative` or `candidate evidence` with parent links | accepted design and implementation evidence collapse |
| Knowledge Graph current status | required menu entry; review evidence does not prove canonical graph implementation | hidden generic fixture, derived artifact, or missing | List it but mark status/provenance; list-first fallback on mobile | derived graph could be mistaken for truth |
| Automatic context capture depth | required fields include visible state; capture tooling may be partial | metadata only, DOM, screenshot, or all | Attach exact available references and label missing fields UNKNOWN | privacy leakage or fabricated completeness |
| Task creation timing | explicit conversion required; start behavior unspecified | create in todo/ready, or create+claim | default to create without execution; offer separately governed `Create and start` only if accepted | task could execute unexpectedly |
| Mobile approval action | high-risk decisions on small screens | full action allowed or review-only | allow only after full evidence/target summary and policy permits; otherwise review-only | accidental approval |
| Public Video nested analysis | handoff says final public behavior needs truthful validation | supported, degraded, or absent | label UNKNOWN/UNVERIFIED until live route evidence exists | candidate overclaims capability |
| Revision within scope | correction may or may not alter accepted Requirements | continue same Task or create new Task/review | require explicit scope comparison; changed acceptance returns to Requirements review | work proceeds under stale authority |
| Surface persistence | close/reopen behavior required, backing state unknown | layout-local state or durable workspace state | preserve object link/version; exact layout persistence can remain local until specified | reopening wrong artifact or losing context |

### Ten comprehension statements

1. I understand that HFA conversation is the entry for interpretation, not a silent Task or approval creator.
2. I understand that BrowserAgent is a workspace profile inside one canonical HFA shell, not a second application shell.
3. I understand that Hermes Profiles own persistent Agent identity and that delegation is a different, bounded primitive.
4. I understand that Hermes Kanban owns durable Task, Attempt, heartbeat, retry, block, and completion truth.
5. I understand that a fresh heartbeat and concrete activity are required before showing ACTIVE.
6. I understand that Proposal acceptance, runtime approval, merge authorization, and deployment authorization are distinct decisions.
7. I understand that Knowledge explains durable meaning while Evidence proves what happened; they remain separate and cross-linked.
8. I understand that Prototype, Browser, Video, Graph, Compare, and surfaces are views/artifacts/adapters, not replacement authorities.
9. I understand that revision history and per-Attempt evidence must be appended and preserved, never silently overwritten.
10. I understand that no implementation begins until both text-first design lanes pass independent audit and explicit acceptance.

### Five rejected design choices

1. Reject “one permanent column per capability”: it destroys usable width, especially with HFA and splits, and treats objects as modes.
2. Reject technical `New surface`: it asks users to manage window mechanics instead of choosing a job; use `Open capability` and focus/reuse intelligently.
3. Reject chat-to-Task auto-conversion: ambiguous intent would become execution without accepted scope or authority.
4. Reject a ServStack shadow Task/Run/approval engine: it would conflict with Hermes truth and create irreconcilable states.
5. Reject one combined “Ship” decision: merge and deployment target different systems, risks, evidence, environments, and rollback plans.

### Five current-prototype mistakes and prevention

1. Collapsed HFA did not restore content width. Prevention: remove the HFA grid track when collapsed and lock full-width behavior as a critical layout contract.
2. Active workspace was unclear. Prevention: one shell header always shows workspace plus Project/Journey/Topic and source freshness.
3. BrowserAgent toolbar was overloaded and used technical surface creation. Prevention: grouped launcher, local object actions, and Advanced disclosure.
4. Generic SurfaceFrame/StickyToolbar/SurfaceBody grid produced blank space and duplicate chrome. Prevention: one shell header, one local surface header, content rows sized by content/minmax rather than reserved generic tracks.
5. Agents, Tasks/Runs, Approvals, Knowledge, and Evidence were insufficiently discoverable. Prevention: named launcher destinations, contextual links, command search, and truthful empty/permission states without permanent columns.

### Story-to-authority traceability

`Thin read adapter` below means an Owner-facing read/navigation seam only; it does not claim a selected implementation module. Exact seams remain coordinated with Issue #64 Track B.

| Story | Menu entry | Canonical objects | Hermes primitive / authority | Extension seam | Evidence | Required test |
|---|---|---|---|---|---|---|
| S1 orientation | Overview | Project/Journey/Topic | ServStack project context; no Hermes substitute claimed | Owner-facing context read model | source/version/freshness | opens correct context; disconnected/unknown truthful |
| S2 Agent | Agents | Profile + Attempts | Hermes Profile; Kanban/session activity | thin read adapter | profile config/status source + run links | configured vs active separated; stale not ACTIVE |
| S3 Task progress | Tasks & Runs | Task + Attempt | Hermes Kanban | Kanban read/action adapter | events, heartbeat, outcomes | Attempt requires Task; freshness fail-closed |
| S4 approval | Approvals | named Approval/Decision | Hermes policy or proper GitHub/Owner authority | authority-routing view | request, target, decision, rollback | Approval ≠ execution; wrong authority cannot act |
| S5 Knowledge | Knowledge | canonical Document | governed wiki/repo source | search/read adapter | source path/version/status | Knowledge source traceable; not compact memory |
| S6 audit | Evidence/Audit | Evidence + Audit | Hermes logs/Kanban and Git/PR/CI/artifacts | append/index view | exact IDs, hashes, reviewer verdict | prior evidence immutable; disputed state visible |
| S7 Browser+Prototype | Browser, Prototype | live route + artifact | browser observation + artifact store | surface composition | route/DOM/console + artifact binding | pair keeps identities; narrow layout switches safely |
| S8 Req+Video | Requirements, Video | Requirement + media | accepted requirement source + media artifact | related-object surface | version/binding/playback state | close surface does not delete; reopen same artifact |
| S9 improve HFA | HFA, Requirements, Tasks & Runs | Conversation→Proposal→Requirement→Task→Attempt | Hermes session/Profile/Kanban for runtime | non-executing proposal + explicit Kanban action | origin links, events, candidate, audit | chat creates no Task; explicit readback required |
| S10 improve BrowserAgent | HFA, Prototype | same chain scoped to BrowserAgent | same Hermes primitives | context envelope + proposal review | route/surface/project refs | correct workspace context; no auto-execution |
| S11 defect revision | Prototype, Evidence | revision request + versions + Attempt | Kanban continuation/new Task only when governed | append-only revision links | before/after artifacts | prior version/evidence retained; scope change reopens review |
| S12 corrected decision | Compare, Evidence, Approvals | candidate + Audit + decisions | Git/PR/CI + Owner/production authority | compare and authority-routing view | exact head, tests, visual evidence, audit | merge does not deploy; exact target/environment required |

### Cross-lane invariants for acceptance

- Conversation != Task.
- Profile != Task and persistent Profile != ephemeral delegation.
- Task != Attempt; Attempt cannot exist without Task reference.
- Approval != execution; merge != deployment.
- Knowledge != Evidence.
- Surface/view != canonical object.
- No fresh heartbeat means no ACTIVE claim.
- Prior evidence and revisions cannot be silently overwritten.
- ServStack may present and cross-link Hermes truth but must not duplicate its runtime engines.
- UI implementation scope must derive from accepted stories/object authority, not from current component names.

### Stop statement

No code, CSS, tests, documentation in the repository, runtime configuration, profile, Skill, Hook, Plugin, MCP, gateway, cron, build, deployment, merge, Production, Hermes Core, preserved `/tmp` evidence, or `/root/hermes-factory` was modified by this packet. The packet awaits Owner/text audit; it does not authorize implementation.

M1_3_IA_UX_COMPREHENSION_REVIEW_READY
